### PR TITLE
Add the datastore API for determining audience type (default, SK-created or user-created).

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #8129 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/datastore/__fixtures__/available-audiences.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[
```
Comment (techanvil) [inline assets/js/modules/analytics-4/datastore/__fixtures__/available-audiences.json:1]: This file is copied in from the [PR](https://github.com/google/site-kit-wp/pull/8606/files#diff-14a74e8381f52f6f42e160be1dff0079e1d0e8bbc7b222bb035f49a1f42621ab) for #8487, I haven't branched from that branch as there are pending changes to apply to the file there.

**Review by hussain-t — APPROVED** [review]
Great work, @techanvil. LGTM 👍

**Review by hussain-t — CHANGES_REQUESTED** [review]
@techanvil, could you resolve the conflicts?

**Comment by techanvil** [comment]
Thanks @hussain-t, the merge conflicts are resolved :)

**Comment by techanvil** [comment]
@hussain-t, there are a couple of failing E2E tests, but these are just the usual flaky tests and aren't related to the changes in this PR. I'm sending back your way rather than rerunning these.

**Review by hussain-t — APPROVED** [review]
Great work, @techanvil. LGTM 👍

**Review by nfmohit — APPROVED** [review]
LGTM ✅

**Pull request merged**